src/Microsoft.FeatureManagement/IContextualFeatureVariantAssigner.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// Provides a method to assign a variant of a feature to be used based off of custom conditions.
+    /// </summary>
+    /// <typeparam name="TContext">A custom type that the assigner requires to perform assignment</typeparam>
+    public interface IContextualFeatureVariantAssigner<TContext> : IFeatureVariantAssignerMetadata
+    {
+        /// <summary>
+        /// Assign a variant of a feature to be used based off of customized criteria.
+        /// </summary>
+        /// <param name="variantAssignmentContext">Information provided by the system to be used during the assignment process.</param>
```

Comment:
System - do you mean "feature management system"?

Reply:
Yes, but I think this should be reworded further to be more consistent with the summary for `IFeatureFilter`

`A variant assignment context that contains information needed to assign a variant for a feature.`

Reply:
Sounds good